### ISSUE TITLE
Bug: false positive for key-spacing with align mode: minimum and afterColon: true

### ISSUE THREAD

**Issue by robopzet** [eslint/eslint#15433]
### Environment

Node version: v14.18.2
npm version: v6.14.15
Local ESLint version: Not found
Global ESLint version: Not found
Operating System: win32 10.0.19044

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
"rules": {
        "key-spacing": ["error", {
            "align": {"beforeColon": true, "afterColon": true, "on": "colon", "mode": "minimum"}
        }]
    }
```
</details>

```js
const ENCODINGS = {
    UTF8 : 'utf8'
}
```

### What did you expect to happen?

No error for key-spacing rule.

### What actually happened?

This error was produced:

    2:9  error  Extra space after key 'UTF8'                    key-spacing

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I've tried several configurations for this rule, including this

    "multiline": {"beforeColon": true, "afterColon": true, "align": "colon", "mode": "minimum"}

From the documentation I think this is allowed, but it produces an error:

> Oops! Something went wrong! :(
> 
> ESLint: 8.4.1
> 
> Error: .eslintrc.json:
>         Configuration for rule "key-spacing" is invalid:
>         Value {"align":{"beforeColon":true,"afterColon":true,"on":"colon","mode":"minimum"},"multiline":{"beforeColon":true,"afterColon":true,"align":"colon","mode":"minimum"}} should NOT have additional properties.
>         Value {"align":{"beforeColon":true,"afterColon":true,"on":"colon","mode":"minimum"},"multiline":{"beforeColon":true,"afterColon":true,"align":"colon","mode":"minimum"}} should NOT have additional properties.
>         Value {"align":{"beforeColon":true,"afterColon":true,"on":"colon","mode":"minimum"},"multiline":{"beforeColon":true,"afterColon":true,"align":"colon","mode":"minimum"}} should NOT have additional properties.
>         Value {"align":{"beforeColon":true,"afterColon":true,"on":"colon","mode":"minimum"},"multiline":{"beforeColon":true,"afterColon":true,"align":"colon","mode":"minimum"}} should match some schema in anyOf.
>

**Comment by aladdin-add** [comment]
Thanks for the report!  I was able repro it in [the demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQga2V5LXNwYWNpbmc6IFtcImVycm9yXCIsIHtcbiAgICBcImFsaWduXCI6IHtcbiAgICAgICAgXCJiZWZvcmVDb2xvblwiOiB0cnVlLFxuICAgICAgICBcImFmdGVyQ29sb25cIjogdHJ1ZSxcbiAgICAgICAgXCJvblwiOiBcImNvbG9uXCJcbiAgICB9XG59XSovXG5jb25zdCBFTkNPRElOR1MgPSB7XG4gICAgVVRGOCA6ICd1dGY4J1xufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibm8tZ2xvYmFsLWFzc2lnbiI6Miwibm8tZXh0ZW5kLW5hdGl2ZSI6MiwiYXJyYXktZWxlbWVudC1uZXdsaW5lIjoyfSwiZW52Ijp7ImJyb3dzZXIiOnRydWUsIm5vZGUiOnRydWV9fX0=).

looks a bug to me.

**Comment by mdjermanovic** [comment]
`align` options apply only to groups of properties. A group is two or more consecutive properties not separated with blank lines.

> ```js
> const ENCODINGS = {
>     UTF8 : 'utf8'
> }
> ```

In this example, there is only one property, so this isn't a group and therefore top-level properties apply. Default for `"beforeColon"` is `false`, that's why rule reports `Extra space after key 'UTF8'`.

Would this configuration work for you?

```js
"key-spacing": ["error", {
            "beforeColon": true,
            "afterColon": true,
            "align": {"beforeColon": true, "afterColon": true, "on": "colon", "mode": "minimum"}
        }]
```

Or, if you want `"mode": "minimum"` to apply to all properties, it can be simplified to:

```js
"key-spacing": ["error", {
            "beforeColon": true,
            "afterColon": true,
            "align": "colon", 
            "mode": "minimum"
        }]
```

**Comment by robopzet** [comment]
Sorry for the report, I guess I didn't read the documentation well enough.

There are quite a lot of rules and it took me some time to create a complete ruleset to work with. When evaluating it I will try this suggestion.
